src/superphot_plus/fit_numpyro.py

```diff
@@ -161,24 +156,47 @@ def jax_model(t=None, obsflux=None, uncertainties=None, max_flux=None, inc_band_
             (1 - sigmoid) * (1 - beta * phase)
             + sigmoid * (1 - beta * gamma) * jnp.exp(-(phase - gamma) / tau_fall)
         )
-
-        # g band
-        phase_b = (t - t0_b)[inc_band_ix]
-        flux_const_b = A / (1.0 + jnp.exp(-phase_b / tau_rise_b))
-        sigmoid_b = 1 / (1 + jnp.exp(10.0 * (gamma_b - phase_b)))
-
-        flux = flux.at[inc_band_ix].set(
-            flux_const_b
-            * (
-                (1 - sigmoid_b) * (1 - beta_b * phase_b)
-                + sigmoid_b * (1 - beta_b * gamma_b) * jnp.exp(-(phase_b - gamma_b) / tau_fall_b)
+        sigma_tot = jnp.sqrt(uncertainties**2 + extra_sigma**2)
+        
+        # auxiliary bands
+        for b_idx, uniq_b in enumerate(unique_bands):
+            if uniq_b == ref_band:
+                continue
+                
+            b_priors = priors.bands[uniq_b]
+            A_ratio = numpyro.sample("A_"+uniq_b, trunc_norm_fields(b_priors.amp))
+            beta_ratio = numpyro.sample("beta_"+uniq_b, trunc_norm_fields(b_priors.beta))
+            gamma_ratio = numpyro.sample("gamma_"+uniq_b, trunc_norm_fields(b_priors.gamma))
+            t0_ratio = numpyro.sample("t0_"+uniq_b, trunc_norm_fields(b_priors.t_0))
+            tau_rise_ratio = numpyro.sample("tau_rise_"+uniq_b, trunc_norm_fields(b_priors.tau_rise))
+            tau_fall_ratio = numpyro.sample("tau_fall_"+uniq_b, trunc_norm_fields(b_priors.tau_fall))
+            extra_sigma_ratio = numpyro.sample("extra_sigma_"+uniq_b, trunc_norm_fields(b_priors.extra_sigma))
+
+            A_b = A * A_ratio  # pylint: disable=unused-variable
+            beta_b = beta * beta_ratio
+            gamma_b = gamma * gamma_ratio
+            t0_b = t0 * t0_ratio
+            tau_rise_b = tau_rise * tau_rise_ratio
+            tau_fall_b = tau_fall * tau_fall_ratio
+
+
+            inc_band_ix = np.arange(b_idx*PAD_SIZE,(b_idx+1)*PAD_SIZE)
```

Comment:
If you want to generalize away from needing to know `PAD_SIZE` in this function, you should be able to replace `inc_band_idx` with something like:

`inc_band = bdata == b_idx`

and adapt the code below like:

`flux[inc_band] = flux_const_b * ...`

Reply:
Unfortunately JAX/JIT really doesn't like boolean indexing within jit-compiled functions, so I can't do that :(